### PR TITLE
test: use Travis CI to run tests on every pull request

### PR DESCRIPTION
**Pull request by cclauss**
Use flake8 to automate the discovery of Python syntax errors and undefined names.

This is a second attempt at #1336 which got into a bad git-state...

##### Checklist

- [ ] `npm install && npm test` passes
- [x] tests are included 
- [ ] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

##### Description of change

There are Python 3 syntax errors and many undefined names which may raise NameError at runtime. This PR runs flake8 runs in two passes: The first looks at critical issues in stop-the-build mode and the second looks at style violations in everything-is-a-warning mode.

### CODE REVIEW

**Comment by rvagg** [comment]
I don't know how to hook this up and don't think I have permissions either. @nodejs/automation can you help?

**Comment by cclauss** [comment]
Someone who has admin rights would need to go to https://travis-ci.org/nodejs/node-gyp and flip the repo switch on.

**Comment by rvagg** [comment]
It's not quite that simple, or at least it never used to be, because of a policy about giving org privs to external services. There's hoops that need to be jumped through to get the github-bot to react to changes in this repo and trigger Travis runs.

**Comment by cclauss** [comment]
Is there an internal test runner available?  Jenkins or something?

**Comment by targos** [comment]
I enabled it. We have the GitHub app so the url is https://travis-ci.com/nodejs/node-gyp

**Comment by targos** [comment]
@rvagg it's simpler since some time with github app because we can give permissions to Travis on a per repository basis (it's in the organization's config)

**Comment by cclauss** [comment]
Thanks much!!  Build in progress... https://travis-ci.com/nodejs/node-gyp/pull_requests

**Comment by targos** [comment]
Shouldn't it end up in a failed state instead of errored?

**Comment by cclauss** [comment]
The tests now pass with one __noqa TODO__ added.  I am unclear what is the difference between failed and errored but I do know that all these issues and more for both Python 2 and Python 3 were flattened in https://github.com/refack/GYP

**Comment by targos** [comment]
I think that anything that fails outside of the "script" phase results in an error.
Is it necessary to run `flake8` before `npm install`? Otherwise I would do:
- `pip install` and `npm install` in the "install" phase
- `flake8` and `npm test` in the "script" phase

**Comment by cclauss** [comment]
Bump on this one please.

**Comment by rvagg** [comment]
👍 will merge this if you fix up the commits, make it 2 or 3 commits - one for travis, one or two for the python fixes as you see fit.

**Comment by rvagg** [comment]
also, there's a "TODO" in there with noqa, is that something that needs to be resolved prior to landing this?

**Comment by cclauss** [comment]
I am OK with leaving the TODO open as it has been working without issue to date.  Sometimes that situation can be caused by implicit imports.

File: gyp/pylib/gyp/input.py [hunk 1]
```diff
@@ -1182,7 +1182,7 @@ def LoadVariablesFromVariablesDict(variables, the_dict, the_dict_key):
       if variable_name in variables:
         # If the variable is already set, don't set it.
         continue
-      if the_dict_key is 'variables' and variable_name in the_dict:
+      if the_dict_key == 'variables' and variable_name in the_dict:
```
Comment (evenstensberg) [inline gyp/pylib/gyp/input.py]: why is this needed?

Comment (cclauss) [reply gyp/pylib/gyp/input.py]: Identity is not the same thing as equality in Python.

Proof:
```
>>> the_dict_key = 'variable'
>>> the_dict_key += 's'
>>> the_dict_key == 'variables'
True
>>> the_dict_key is 'variables'
False
>>>
```

**Comment by rvagg** [comment]
landed in 051b6ed
